.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,185 @@
+name: Pull Request
+on: [pull_request, workflow_dispatch]
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Heads up, this may break if the PR was created using a fork, which I think may be quite common considering it's open to the public. `secrets` cannot be used in PRs created by forks as a security design, otherwise anyone without push access to the repo would be able to extract information from secrets.

I haven't found an elegant way around this, and I don't think github provides anything ideal either. The only way I've been able to circumnavigate this is by adding conditionals to any steps that use secrets:
`if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository`

This will resolve to true only if the event was not triggered from a PR, OR if the feature branch is located in the origin repo. Since this workflow is just for PRs, you can likely remove the first part of this condition.

Reply:
This already works on master, because GITHUB_TOKEN is provided by default - it's not actually a "secret" that we set.

Reply:
> This already works on master, because GITHUB_TOKEN is provided by default - it's not actually a "secret" that we set.

This workflow shouldn't run on `master` due to the `on` configuration (unless someone ran it against master via `workflow_dispatch`). This workflow will only be automatically triggered on pull requests, which includes PRs created from forks. I was able to test this in other repos by forking them, and creating a PR from my fork to a branch in the origin.

Reply:
Sorry, I didn't mean that the workflow is meant to run on master. I just meant that we already have this exact same job/syntax in our existing workflows, so we don't need to worry about it not working.

Reply:
Oh I see what you meant. I should've placed this comment under a different area because you're right, it doesn't apply to `secrets.GITHUB_TOKEN`, but this will definitely break on forked PRs for any other secret. Special logic should be put in place for any steps that use secrets to ensure they do not run on forked PRs (unless the step only uses `secrets.GITHUB_TOKEN`).